k8srm-prototype/pod-spec-examples.yaml

```diff
@@ -0,0 +1,127 @@
+# not sure that's better.
+
+
+# Allocates four SR-IOV VFs from any Intel NIC These could even each come from
+# a different NIC (on the same node of course). NICs are not tied to specific
+# containers, so there is no need to name the claim.
+deviceClaims:
+- template:
+    claims:
+    - deviceClass: "intel-sriov-vf"
+      requests:
+        count: 4
+
+# Allocates four SR-IOV VFs from any vendor NIC, making sure all the VFs are on
+# the same PF this requires vendor & PF together to fully qualify a physical
+# card on a node.
+deviceClaims:
+- template:
+    claims:
+    - deviceType: "sriov-vf"
+      count: 4
+      matchAttributes:
+      - "vendor"
+      - "pf"
+
+# Allocates two A100 GPUs on the same PCI root complex. The name is used to
+# attach the associated claims to a container.
+deviceClaims:
+- template:
+    claims:
+    - name: gpus
+      deviceClass: "nvidia-a100"
+      count: 2
+      matchAttributes:
+      - pcie-root
+
+# Allocates two A100 GPUs on the same PCI root complex, but keeps them
+# separated in the API so they can be attached to two differenti containers.
+deviceClaims:
+- template:
+    matchAttributes:
+    - pcie-root
+    claims:
+    - name: gpu-one
+      deviceClass: "nvidia-a100"
+    - name: gpu-two
+      deviceClass: "nvidia-a100"
+
+# Allocates an SR-IOV PF and a an A100 from the same pcie-root. Name is
+# populated here for the PF just to show that it can be, but the expectation is
+# generally for network devices it is not needed. Note ANY vendor PF is fine
+# here; we use deviceType, not deviceClass.
+deviceClaims:
+- template:
+    matchAttributes:
+    - pcie-root
+    claims:
+    - name: gpu
+      deviceClass: "nvidia-a100"
+    - name: nic
+      deviceType: "sriov-pf"
+
+# Allocates either an A100, H100, or two L4s that are on the same pcie root.
+deviceClaims:
+- template:
+    claims:
+    - oneOf:
+      - deviceClass: "nvidia-a100"
+      - deviceClass: "nvidia-h100"
+      - deviceClass: "nvidia-l4"
+        count: 2
+        matchAttributes:
+        - pcie-root
+
+# Allocates and SR-IOV VF from any vendor and either an A100, H100, or two L4s,
+# all from the same NUMA node and PCIe root complex.
+deviceClaims:
+- template:
+    matchAttributes:
+    - numa
+    - pcie-root
+    claims:
+    - deviceType: "sriov-vf"
+    - oneOf:
+      - deviceClass: "nvidia-a100"
+      - deviceClass: "nvidia-h100"
+      - deviceClass: "nvidia-l4"
+        count: 2
+
+# Allocates 4 VFs from 4 different PFs,
+# but all the same vendor
+deviceClaims:
+- template:
+  claims:
+  - deviceType: "sriov-vf"
+    count: 4
+    matchAttributes:
+    - vendor
+    distinctAttributes: # maybe we should use affinity/antiaffinity terms?
+    - pf
+
+# Allocates a partial H100 in MPS mode, declaring a reservation of 8GB of
+# memory to regulate the number of clients that attach.
+deviceClaims:
+- template:
+  claims:
+  - deviceClass: "nvidia-h100-mps"
```

Comment:
Likewise with MPS -- I don't expect the admin to create a class for that. It will be a configuration parameter in the request claim.

Reply:
So, I am getting the opposite sort of feedback from others. That users will have little understanding of these details, and that platform admins will curate a set of pre-configured classes. Which means we need to move *more* to the class, and let the claim just refer to that. We should ask more in the WG to find out what our end user and platform admin representatives think.